flo-runner/src/main/java/com/spotify/flo/context/ForkingEvalContext.java

```diff
@@ -0,0 +1,385 @@
+      log.debug("Forking forcibly enabled (environment variable FLO_FORCE_FORK=true)");
+      return new ForkingEvalContext(baseContext);
+    } else if (DISABLE_FORKING) {
+      log.debug("Forking disabled (environment variable FLO_DISABLE_FORKING=true)");
+      return baseContext;
+    } else if (IN_DEBUGGER) {
+      log.debug("In debugger, forking disabled (enable by setting environment variable FLO_FORCE_FORK=true)");
+      return baseContext;
+    } else {
+      return new ForkingEvalContext(baseContext);
+    }
+  }
+
+  @Override
+  public <T> Value<T> value(Fn<T> value) {
+    return super.value(fork(value));
+  }
+
+  private <T> Fn<T> fork(Fn<T> value) {
+    return () -> {
+
+      final ExecutorService executor = Executors.newCachedThreadPool();
+      Path tempdir = null;
+
+      try {
+        final Path workdir;
+        try {
+          tempdir = Files.createTempDirectory("flo-fork");
+          workdir = Files.createDirectory(tempdir.resolve("workdir"));
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+
+        final Path closureFile = tempdir.resolve("closure");
+        final Path resultFile = tempdir.resolve("result");
+        final Path errorFile = tempdir.resolve("error");
+
+        final String home = System.getProperty("java.home");
+        final String classPath = System.getProperty("java.class.path");
+        final Path java = Paths.get(home, "bin", "java").toAbsolutePath().normalize();
+
+        log.debug("serializing closure");
+        try {
+          PersistingContext.serialize(value, closureFile);
+        } catch (Exception e) {
+          throw new RuntimeException("Failed to serialize closure", e);
+        }
+
+        final ProcessBuilder processBuilder = new ProcessBuilder(java.toString(), "-cp", classPath)
+            .directory(workdir.toFile());
+
+        // Propagate -Xmx.
+        // Note: This is suboptimal because if the user has configured a max heap size we will effectively use that
+        // times the concurrent nummber of executing task processes in addition to the heap of the parent process.
+        // However, propagating a lower limit might make the task fail if the user has supplied a heap size that is
+        // tailored to the memory requirements of the task.
+        ManagementFactory.getRuntimeMXBean().getInputArguments().stream()
+            .filter(s -> s.startsWith("-Xmx"))
+            .forEach(processBuilder.command()::add);
+
+        // Trampoline arguments
+        processBuilder.command().add(Trampoline.class.getName());
+        processBuilder.command().add(closureFile.toString());
+        processBuilder.command().add(resultFile.toString());
+        processBuilder.command().add(errorFile.toString());
+
+        // Propagate TASK_ID to child process
+        final TaskId taskId = Tracing.TASK_ID.get();
+        if (taskId != null) {
+          processBuilder.environment().put("FLO_TASK_ID", taskId.toString());
```

Comment:
I hope Java by default carries over all environment to subprocess.

Reply:
It does.